src/dependency-manager/src/service-config/v1.ts

```diff
@@ -51,38 +74,63 @@ class ApiSpecV1 {
   liveness_probe?: LivenessProbeV1;
 }
 
+class InterfaceSpecV1 {
+  description?: string;
+  host?: string;
+  port!: number;
+}
+
 export class ServiceVolumeV1 {
-  mountPath?: string;
+  mountPath?: string;  // TODO: TJ Ask David about volumes camelcase
+  hostPath?: string;
   description?: string;
   readonly?: boolean;
 }
 
-export class ServiceVolumesV1 {
-  [s: string]: ServiceVolumeV1;
+class ServiceDebugOptionsV1 {
+  path?: string;
+  dockerfile?: string;
+  @Transform(value => (transformVolumes(value)))
+  volumes?: { [s: string]: ServiceVolumeV1 };
+  command?: string | string[];
+  entrypoint?: string | string[];
+}
+
+interface IngressSpecV1 {
+  subdomain: string;
 }
 
 export class ServiceConfigV1 extends ServiceConfig {
   __version = '1.0.0';
-  name = '';
-  description?: string;
-  keywords?: string[];
-  image?: string;
-  port?: string;
-  command?: string | string[];
-  entrypoint?: string | string[];
-  dependencies: { [s: string]: string } = {};
-  language?: string;
-  debug?: string;
-  parameters: { [s: string]: ServiceParameterV1 } = {};
-  datastores: { [s: string]: ServiceDatastoreV1 } = {};
+  protected name?: string;
```

Comment:
TODO: Remove protected - they won't be expose unless we explicitly type as ServiceConfigV1 and we need access to edit replicas, parameters, etc. in the UI.